CMakeLists.txt

```diff
@@ -260,6 +260,8 @@ if (LIBDDWAF_BUILD_SHARED)
             else()
                 add_custom_command(TARGET libddwaf_shared POST_BUILD
                     COMMAND ${STRIP} -S -x $<TARGET_FILE:libddwaf_shared>)
+                add_custom_command(TARGET libddwaf_static POST_BUILD
```

Comment:
There's no guarantee that the static library will be built here, notice that this is inside an if statement `if (LIBDDWAF_BUILD_SHARED)`.

This should be done above inside https://github.com/DataDog/libddwaf/pull/107/files#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20aR149

Also the debug symbols should be stored as well.

Reply:
Storing debug symbols for static libraries on MacOs seems to be quite a painful process.
I managed to do so by extracting all .o files from the unstripped .a and invoking `dsymutil` on those, however dsymutil seems to complain that there's no debug symbol to be found, which can be confirmed by calling `dsymutil -dump-debug-map` on the generated .dwarf files.
FYI here is how I configure cmake and build:
` cmake .. -DCMAKE_BUILD_TYPE=RelWithDebInfo -Wno-dev`
 `make -j8`
 
 @Anilm3 have you had any experience with this?

Reply:
Unfortunately, I don't have any advice, I'm surprised that the error is that the symbols are missing. Perhaps @lloeki has advice as he has more experience with MacOs.

Alternatively, as a measure of last resort if you find that this can't be achieved, consider keeping both stripped and original versions of the `.a`.